File: Interview_Prep__App/native_java/src/main/java/com/dreams/androidquizapp/network/Tc2rGithubService.java
package com.dreams.androidquizapp.network;

import com.dreams.androidquizapp.repository.reponseModels.AnswersResponse;
import com.dreams.androidquizapp.repository.reponseModels.QuestionsResponse;

import retrofit2.Call;
import retrofit2.http.GET;

public interface Tc2rGithubService {
    // Request method and URL specified in the annotation

    @GET("master/Languages/Android/answers.json")
    Call<AnswersResponse> getAnswersList();

    @GET("master/Languages/Android/questions.json")
    Call<QuestionsResponse> getQuestionsList();
}

File: Interview_Prep_App/native_kotlin/view_binding/src/main/java/com/interviewprep/kotlinviewbinding/repository/tc2rgithubrepository/source/remote/Tc2rGithubService.kt
package com.interviewprep.kotlinviewbinding.repository.tc2rgithubrepository.source.remote

import com.interviewprep.kotlinviewbinding.repository.models.response.AnswersResponse
import com.interviewprep.kotlinviewbinding.repository.models.response.QuestionsResponse
import retrofit2.Response
import retrofit2.http.GET

internal interface Tc2rGithubService {
    // Request method and URL specified in the annotation

    @GET("Tc2r1/DevInterview_Questions/master/Languages/Android/answers.json")
    suspend fun requestAnswers(): Response<AnswersResponse>

    // Gets answers objects
    @GET("Tc2r1/DevInterview_Questions/master/Languages/Android/questions.json")
    suspend fun getQuestionsJson(): Response<QuestionsResponse>
}
